dom0-updates/qfile-dom0-unpacker.c

```diff
@@ -49,11 +53,20 @@ int main(int argc, char ** argv)
 	long long files_limit = DEFAULT_MAX_UPDATES_FILES;
 	long long bytes_limit = DEFAULT_MAX_UPDATES_BYTES;
 
+	struct statvfs st;
+	long long root_free_space;
+
 	if (argc < 3) {
 		fprintf(stderr, "Invalid parameters, usage: %s user dir [-v]\n", argv[0]);
 		exit(1);
 	}
 
+	statvfs("/", &st);
```

Comment:
One more thing: if /var/lib/qubes/updates (or whatever the target directory is) is on a separate partition, it should measure that. You can do that easily if you move setting the limit after `mkdir()` call below.

Reply:
Done.